### ISSUE TITLE
Constant HANDSHAKE then DISCONNECT

### ISSUE THREAD

**Issue by Sashweed** [Conchylicultor/DeepQA#43]
Morning Git people,

Is anyone having and handshake issues with the web server?

my architecture looks like this:

Windows 7 desktop running chrome - 10.97.43.87
Ubuntu Server running docker 10.29.108.71
2 docker containers

**6297f2dd8135        sashweed:tensorflow   "/bin/bash"              19 minutes ago      Up 19 minutes             0.0.0.0:32775->8000/tcp   fervent_leakey
cf33127cc4a2        redis                 "docker-entrypoint..."   47 hours ago        Up 17 hours               6379/tcp                  redis**

but when I use my desktop to connect to http://10.29.108.71:32775/ I get served the page but then constant HANDSHAKE then DISCONNECT as you can see below. (and obviously the site doesn't actually do anything)

there are no firewalls between my desktop and the server and the iptables look correct

jsalvin@HAL:~$ sudo iptables -t nat -L -n
[sudo] password for jsalvin:
Chain PREROUTING (policy ACCEPT)
target     prot opt source               destination
DOCKER     all  --  0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL

Chain INPUT (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination
DOCKER     all  --  0.0.0.0/0           !127.0.0.0/8          ADDRTYPE match dst-type LOCAL

Chain POSTROUTING (policy ACCEPT)
target     prot opt source               destination
MASQUERADE  all  --  172.17.0.0/16        0.0.0.0/0
MASQUERADE  tcp  --  172.17.0.2           172.17.0.2           tcp dpt:8000

Chain DOCKER (2 references)
target     prot opt source               destination
RETURN     all  --  0.0.0.0/0            0.0.0.0/0
DNAT       tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:32775 to:172.17.0.2:8000

side note: I'm also getting this without docker when I install both TF, DeepQA and REDIS on same ubuntu server.

###############
actual run logs
###############
root@6297f2dd8135:~/DeepQA/chatbot_website# python3 manage.py runserver 172.17.0.2:8000
Initializing bot...
Welcome to DeepQA v0.1 !

TensorFlow detected: v0.11.0
Loading dataset from /root/DeepQA/data/samples/...
Loaded opensubs: 96895 words, 981660 QA
Model creation...
Initialize variables...
WARNING: No previous model found, starting from clean directory: /root/DeepQA/save/model-server
Daemon mode, running in background...
Initializing bot...
Welcome to DeepQA v0.1 !

TensorFlow detected: v0.11.0
Loading dataset from /root/DeepQA/data/samples/...
Loaded opensubs: 96895 words, 981660 QA
Model creation...
Initialize variables...
WARNING: No previous model found, starting from clean directory: /root/DeepQA/save/model-server
Daemon mode, running in background...
Performing system checks...

System check identified no issues (0 silenced).
January 24, 2017 - 22:59:29
Django version 1.10.5, using settings 'chatbot_website.settings'
Starting Channels development server at http://172.17.0.2:8000/
Channel layer default (asgi_redis.core.RedisChannelLayer)
Quit the server with CONTROL-C.
2017-01-24 22:59:29,487 - INFO - worker - Listening on channels http.request, websocket.connect, websocket.disconnect, websocket.receive
2017-01-24 22:59:29,488 - INFO - worker - Listening on channels http.request, websocket.connect, websocket.disconnect, websocket.receive
2017-01-24 22:59:29,489 - INFO - worker - Listening on channels http.request, websocket.connect, websocket.disconnect, websocket.receive
2017-01-24 22:59:29,491 - INFO - worker - Listening on channels http.request, websocket.connect, websocket.disconnect, websocket.receive
2017-01-24 22:59:29,493 - INFO - server - Using busy-loop synchronous mode on channel layer
2017-01-24 22:59:29,493 - INFO - server - Listening on endpoint tcp:port=8000:interface=172.17.0.2
####### THEN I CONNECT FROM MY DESKTOP
[2017/01/24 22:59:35] HTTP GET / 200 [0.03, 10.97.43.87:53644]
[2017/01/24 22:59:35] HTTP GET /static/assets/css/main.css 200 [0.03, 10.97.43.87:53644]
[2017/01/24 22:59:35] HTTP GET /static/images/avatar.jpg 200 [0.03, 10.97.43.87:53645]
[2017/01/24 22:59:35] HTTP GET /static/js/jquery-3.1.0.min.js 200 [0.04, 10.97.43.87:53645]
[2017/01/24 22:59:35] HTTP GET /static/js/reconnecting-websocket.js 200 [0.03, 10.97.43.87:53644]
[2017/01/24 22:59:35] HTTP GET /static/js/chat.js 200 [0.04, 10.97.43.87:53647]
[2017/01/24 22:59:35] HTTP GET /static/assets/css/font-awesome.min.css 200 [0.04, 10.97.43.87:53644]
[2017/01/24 22:59:35] HTTP GET /static/assets/fonts/fontawesome-webfont.woff2?v=4.6.3 200 [0.04, 10.97.43.87:53647]
[2017/01/24 22:59:35] HTTP GET /static/assets/css/images/overlay.png 200 [0.06, 10.97.43.87:53644]
[2017/01/24 22:59:35] HTTP GET /static/images/bg.png 200 [0.06, 10.97.43.87:53645]
[2017/01/24 22:59:36] WebSocket HANDSHAKING /chat [10.97.43.87:53650]
[2017/01/24 22:59:36] HTTP GET / 200 [0.04, 10.97.43.87:53645]
[2017/01/24 22:59:36] HTTP GET /static/images/icon.png 200 [0.03, 10.97.43.87:53647]
New client connected: room-10.97.43.87-53650
[2017/01/24 22:59:38] WebSocket DISCONNECT /chat [10.97.43.87:53650]
Client disconnected: room-10.97.43.87-53650
[2017/01/24 22:59:39] WebSocket HANDSHAKING /chat [10.97.43.87:53653]
New client connected: room-10.97.43.87-53653
[2017/01/24 22:59:41] WebSocket DISCONNECT /chat [10.97.43.87:53653]
Client disconnected: room-10.97.43.87-53653
[2017/01/24 22:59:42] WebSocket HANDSHAKING /chat [10.97.43.87:53655]
New client connected: room-10.97.43.87-53655
[2017/01/24 22:59:44] WebSocket DISCONNECT /chat [10.97.43.87:53655]
Client disconnected: room-10.97.43.87-53655
[2017/01/24 22:59:47] WebSocket HANDSHAKING /chat [10.97.43.87:53658]
New client connected: room-10.97.43.87-53658
[2017/01/24 22:59:49] WebSocket DISCONNECT /chat [10.97.43.87:53658]
Client disconnected: room-10.97.43.87-53658

any help appreciated

**Comment by Conchylicultor** [comment]
From the logs, it seems the model isn't detected. Are you sure that the models are present on the right folder. If so, can you tell me what is the content of the `model-server` folder.
```
WARNING: No previous model found, starting from clean directory: /root/DeepQA/save/model-server
```

**Comment by jybaek** [comment]
I have the same problem. But I am not a docker.
(main.py) It seems that the default directory is "save/model". So there is nothing in "save/model-server". Is not it? However, copying "save/model" to "save/model-server" still does not work. Is the data too large?

My directory status.
```bash
(tensorflow3) oops@oops:DeepQA$ ll save/model-server/
Total 245880
drwxr-xr-x 2 oops oops      4096  1월 26 09:18 ./
drwxrwxr-x 5 oops oops      4096  1월 26 09:43 ../
-rw-rw-r-- 1 oops oops       163  1월 26 09:18 checkpoint
-rw-rw-r-- 1 oops oops  45620897  1월 26 08:48 events.out.tfevents.1484789286.oops
-rw-rw-r-- 1 oops oops        83  1월 26 08:48 events.out.tfevents.1484801627.oops
-rw-rw-r-- 1 oops oops    472180  1월 26 08:48 events.out.tfevents.1484801949.oops
-rw-rw-r-- 1 oops oops 160086354  1월 26 08:48 model.ckpt
-rw-rw-r-- 1 oops oops  45569359  1월 26 08:48 model.ckpt.meta
-rw-rw-r-- 1 oops oops       264  1월 26 08:48 params.ini
```

**Comment by Sashweed** [comment]
yep looks like your correct, my model-server is empty

when testing this on cornell (I'm 99% sure) when you start the server it copied model to model-server but when I do it with opensubs it just creates the dir model-server but doesn't put anything in there?

**Comment by Conchylicultor** [comment]
Solved with #53 (thanks to @jybaek)
